docs/sql-ref-syntax-qry-select-like.md

```diff
@@ -26,7 +26,7 @@ A LIKE predicate is used to search for a specific pattern.
 ### Syntax
 
 ```sql
-[ NOT ] { LIKE search_pattern [ ESCAPE esc_char ] | RLIKE regex_pattern }
+[ NOT ] { LIKE search_pattern [ ESCAPE esc_char ] | RLIKE regex_pattern | REGEXP regex_pattern}
```

Comment:
`[ RLIKE | REGEXP ] regex_pattern` cc: @huaxingao

Reply:
I think its okay for this PR to just update this doc.

Reply:
have updated and added examples for this usage.

Reply:
Btw, this reminded me that there are still missing keywords in the SQL docs: https://issues.apache.org/jira/browse/SPARK-31753
Its very helpful if you take this over.

Reply:
I am glad to take this over,will work on it later :)

Reply:
Thanks! That's very helpful.